doc: new `persistent` and `transient` keywords (#4792)
dfinity/motoko: doc/md/writing-motoko/writing-intro.md

The new interface is compatible with the previous one, allowing existing clients referencing the canister to continue to work. New clients will be able to exploit its upgraded functionality, in this example the additional `reset` function.

To make it more convenient to declare stable variables, and to prevent missing
`stable` declarations,

> Motoko let's you prefix the entire actor with the keyword `persistent`. In a `persistent` actor,

Reviewer: ```suggestion
Motoko allows you to prefix the entire actor with the keyword `persistent`. In a `persistent` actor,
```

all declarations are `stable` by default. Only declarations that are explicitly marked `transient` will be discarded on upgrade.

``` motoko file=../examples/PersistentStableCounter.mo
```

In this example, `value` is now implicitly stable, while `invocations` is just a transient

> temporary that won't survive upgrades: it counts the number of calls to `inc` since installation.

Author: ```suggestion
temporary that won't survive upgrades: it counts the number of calls to `inc` since the first installation or last upgrade.
```

> temporary that won't survive upgrades: it counts the number of calls to `inc` since the first installation or last upgrade.

Reviewer: ```suggestion
temporary declaration that won't survive upgrades: it counts the number of calls to `inc` since the first installation or last upgrade.
```
